documentation/create_overview_doc/createdoc.sh

```diff
@@ -111,419 +82,215 @@ function read_skip_list() {
-    echo "All $MAX_RETRIES attempts to call bito with prompt file '$filename' have failed to return adequate content. Exiting with error." >&2
+    echo "Failed to call Bito after $MAX_RETRIES attempts. $error_message" >&2
     return 1
 }
 
-# Generate documentation for an individual module
 function create_module_documentation() {
     local path_to_module="$1"
     local documentation_directory="$2"
-    echo "Creating documentation for module: $path_to_module" >&2
+    echo "Generating documentation for the module at $path_to_module..." >&2
 
     if is_skippable "$path_to_module"; then
         echo "Skipped $path_to_module as it's on the exclusion list."
         return
     fi
-    
+
     local name_of_module=$(basename "$path_to_module")
     local content_of_module=$(<"$path_to_module")
 
     local high_level_documentation
-    high_level_documentation=$(call_bito_with_retry "Module: $name_of_module\n---\n$content_of_module" "$prompt_folder/high_level_doc_prompt.txt")
+    high_level_documentation=$(call_bito_with_retry "Module: $name_of_module\n---\n$content_of_module" "$prompt_folder/high_level_doc_prompt.txt" "High-level documentation creation failed for module: $name_of_module")
     local ret_code=$?
     if ! bito_response_ok "$ret_code" "$high_level_documentation"; then
         echo "High-level documentation creation failed for module: $name_of_module"
         return 1
     fi
-    update_token_usage "$content_of_module" "$high_level_documentation"
-
-    local mermaid_diagram=$(create_mermaid_diagram "$name_of_module" "$content_of_module")
-    if [ $? -ne 0 ]; then
-        echo "Mermaid diagram creation failed for module: $name_of_module"
-        return 1
-    fi
 
-    local mdd_file="$documentation_directory/$name_of_module.mdd"
-    if [ ! -s "$mdd_file" ]; then
-        echo -e "$mermaid_diagram" > "$mdd_file"
-    fi
-    update_token_usage "$content_of_module" "$mermaid_diagram"
+    update_token_usage "$content_of_module" "$high_level_documentation"
 
     local markdown_documentation_file="$documentation_directory/${name_of_module}_Doc.md"
     echo -e "## Module: $name_of_module\n$high_level_documentation" >> "$markdown_documentation_file"
-    if [[ -n "$mermaid_diagram" && "$mermaid_diagram" =~ [^[:space:]] ]]; then 
-        echo -e "## Flow Diagram [via mermaid]\n\`\`\`mermaid\n$mermaid_diagram\n\`\`\`" >> "$markdown_documentation_file"
-    fi 
 
-    echo -e "Documentation saved to $markdown_documentation_file\n\n"
+    # Generate the flow chart for the module
+    generate_flow_chart "$path_to_module" "$documentation_directory/flowmaps"
+
+    # Add link to the flow chart in the documentation
+    local flow_chart_name=$(basename "$path_to_module" | cut -f 1 -d '.')
+    echo -e "\n![Flow Chart of $flow_chart_name](./flowmaps/${flow_chart_name}_FlowChart.png)\n" >> "$markdown_documentation_file"
+
+    echo -e "Documentation saved to $markdown_documentation_file\n\n" >&2
```

Comment:
**Issue**: The documentation update process now includes a link to the generated flowchart, but there is no check to ensure that the flowchart was successfully generated before adding the link. <br> **Fix**: Add a check to verify that the flowchart file exists and is not empty before adding the link to the documentation. <br> **Code Suggestion**: 
 ```
 +    if [ -s "./flowmaps/${flow_chart_name}_FlowChart.png" ]; then
 +        echo -e "\\n![Flow Chart of $flow_chart_name](./flowmaps/${flow_chart_name}_FlowChart.png)\\n" >> "$markdown_documentation_file"
 +    else
 +        echo "Failed to generate flowchart for $flow_chart_name" >&2
 +    fi
 ```